pkg/services/resource.go

```diff
@@ -154,6 +155,41 @@ func (s *sqlResourceService) UpdateStatus(ctx context.Context, resource *api.Res
 		return found, nil
 	}
 
+	resourceStatusJSON, err := json.Marshal(resource.Status)
+	if err != nil {
+		return nil, errors.GeneralError("Unable to marshal resource status: %s", err)
+	}
+	resourceStatus := &api.ResourceStatus{}
+	if err := json.Unmarshal(resourceStatusJSON, resourceStatus); err != nil {
+		return nil, errors.GeneralError("Unable to unmarshal resource status: %s", err)
+	}
+
+	foundStatusJSON, err := json.Marshal(found.Status)
+	if err != nil {
+		return nil, errors.GeneralError("Unable to marshal resource status: %s", err)
+	}
+	foundStatus := &api.ResourceStatus{}
+	if err := json.Unmarshal(foundStatusJSON, foundStatus); err != nil {
+		return nil, errors.GeneralError("Unable to unmarshal resource status: %s", err)
+	}
```

Comment:
nit: cloud we add a help function to convert the status object?

Reply:
add helper function and testings.